src/store/store.js

```diff
@@ -2,82 +2,112 @@ import {makeAutoObservable} from 'mobx';
 
 class Store {
   BASE_URL = 'http://192.168.100.106:1339';
+
   categories = [];
+
   currentCategory = [];
+
   questionSets;
+
   chosenQuestionsSet = [];
+
   questions;
+
   currentQuestion = [];
+
   currentAnswers = [];
+
   chosenMode = '';
+
   showHelpDialog = false;
+
   isChooseModeDialog = false;
 
+  isAnswerHasChosen = false;
+
   constructor() {
-    (async () => {
-      await this.fetchCategories();
-      await this.fetchQuestionSets();
-      await this.fetchQuestions();
-    })();
+    this.fetchCategories().catch((error) => console.log('fetchCategories error ', error));
+    this.fetchQuestionSets().catch((error) => console.log('fetchQuestionSets error ', error));
+    this.fetchQuestions().catch((error) => console.log('fetchQuestions error', error));
     makeAutoObservable(this);
   }
 
-  setChosenMode = (chosenMode) => this.chosenMode = chosenMode;
+  setIsAnswerHasChosen = (bool) => {
+    this.isAnswerHasChosen = bool;
+  };
+
+  setChosenMode = (chosenMode) => {
+    this.chosenMode = chosenMode;
+  };
 
-  setCurrentQuestion = (question) => this.currentQuestion = question;
+  setCurrentQuestion = (question) => {
+    this.currentQuestion = question;
+  };
 
   setCurrentAnswers = () => {
     const [{answers}] = this.questions.filter((question) => question.text === this.currentQuestion.text);
-    const [{createdAt, id, updatedAt, ...answersArr}] = answers;
+    const [{
+      createdAt, id, updatedAt, ...answersArr
```

Comment:
maybe I'm wrong, but you are getting a first item out of answers array, then why is it called answersArr? And what is answers variable in this case? Maybe answersArr -> answer?